index.js

```diff
@@ -407,11 +407,16 @@ let rcl = {
   },
 };
 
-process.on('SIGINT', () => {
+const _sigHandler = () => {
   debug('Catch ctrl+c event and will cleanup and terminate.');
-  rcl.shutdown();
-  process.exit(0);
-});
+  // only run when no other signal handlers are installed
```

Comment:
I really like this change and it should replace #721 for these lines. Two things though: The debug call in line 411 should be moved down into the if-check, and maybe we should document how this works in the module documentation? E.g something along:

> Only if no other handlers for `SIGINT` events are registered, *rclnodejs* will shut down all live contexts in these events. This is useful to gracefully exit the process when `CTRL+C` is pressed. Once other signal handlers are registered using `process.on('SIGINT', some_handler)`, the handler of *rclnodejs* will not attempt to shut down the contexts.

Reply:
Alternatively, we could supply an extra argument to specify `auto_cleanup=False` to `rclnodejs.init(conetxt, argv)` to make it explicit. This might cause less confusion for users.